pkg/assembler/backends/testing/backend.go

```diff
@@ -69,6 +69,42 @@ func (c *demoClient) Packages(ctx context.Context, pkgSpec *model.PkgSpec) ([]*m
 	return packages, nil
 }
 
+func (c *demoClient) PackagesType(ctx context.Context, pkgSpec *model.PkgSpec) ([]*model.Package, error) {
```

Comment:
Oh, I was thinking these additional resolvers would be needed only by the neo4j backend so they would be internal there. We won't expose them to the other backends as they are just optimizations in how the neo4j receives the data

Reply:
The `queryResolver` goes to the interface so it would have to be implemented by all the backends

Reply:
Agree, but I was thinking we can make the public resolver then call private functions in the module. Kind of like what we did on ingesting:

https://github.com/guacsec/guac/blob/421309e7419590db05b2a31e5363f88f36183e29/pkg/assembler/backends/testing/ingest_pkg.go#L74-L87

Here `registerPackage` is part of the interface but `registerNamespace` not anymore

Reply:
got it making the change.

Reply:
how about this? Combining via string builder would be messy. This would be easier to modify each individual query as we need.